File: python/gpt/p1.py
def copy_bytes(dest, src, num_bytes):
    dest[:num_bytes] = src[:num_bytes]


src = bytearray(b"Hello, World!")
dest = bytearray(20)

copy_bytes(dest, src, 13)  # Copy 13 bytes (Hello, World!)
print(dest.decode())  # Output: Hello, World!

File: c/gpt/p1.c
#include <stdio.h>
#include <string.h>

void copy_bytes(void *dest, void *src, size_t num_bytes)
{
    memcpy(dest, src, num_bytes);
}

int main()
{
    char src[] = "Hello, World!";
    char dest[20];

    copy_bytes(dest, src, 13); // Copy 13 bytes (Hello, World!)

    printf("Destination: %s\n", dest); // Output: Hello, World!
    return 0;
}
